File: linkedLists/reverseLinkedList.js
// 206. Reverse Linked List
// Easy

// Share
// Given the head of a singly linked list, reverse the list, and return the reversed list.

 

// Example 1:


// Input: head = [1,2,3,4,5]
// Output: [5,4,3,2,1]






var reverseList = function(head) {
    let prev = null;
    while(head) {
        let temp = head;
        head = head.next;
        temp.next = prev;
        prev = temp;
    }
    return prev
}

File: linkedLists/reverseLinkedList.py
# 206. Reverse Linked List
# Easy

# Share
# Given the head of a singly linked list, reverse the list, and return the reversed list.

 

# Example 1:


# Input: head = [1,2,3,4,5]
# Output: [5,4,3,2,1]



# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        prev = None
        while head:
            temp = head
            head = head.next
            temp.next = prev
            prev = temp

        return prev
            
